pioneer/packages/joy-utils/src/transport/tokenomics.ts

```diff
@@ -341,4 +350,133 @@ export default class TokenomicsTransport extends BaseTransport {
+    const proposalCount = (await this.api.query.proposalsEngine.proposalCount() as u32).toNumber();
+    const numberOfForumCategories = (await this.api.query.forum.nextCategoryId() as CategoryId).toNumber() - 1;
+    const numberOfForumPosts = (await this.api.query.forum.nextPostId() as PostId).toNumber() - 1;
+    const [councilMintId, contentCuratorMintId, storageMintId] = await Promise.all([
+      (await this.api.query.council.councilMint()) as MintId,
+      (await this.api.query.contentWorkingGroup.mint()) as MintId,
+      (await this.api.query.storageWorkingGroup.mint()) as MintId
+    ]);
+    const [councilMint, contentMint, storageMint] = await this.api.query.minting.mints.multi<Mint>([
+      councilMintId,
+      contentCuratorMintId,
+      storageMintId
+    ]);
+
+    return {
+      blockHeight,
+      numberOfMembers,
+      content: content.length,
+      numberOfChannels,
+      proposalCount,
+      historicalProposals: HISTORICAL_PROPOSALS.length,
+      numberOfForumCategories,
+      numberOfForumPosts,
+      councilMintCapacity: councilMint.capacity.toNumber(),
+      councilMintSpent: councilMint.total_minted.toNumber(),
+      contentCuratorMintCapacity: contentMint.capacity.toNumber(),
+      contentCuratorMintSpent: contentMint.total_minted.toNumber(),
+      storageProviderMintCapacity: storageMint.capacity.toNumber(),
+      storageProviderMintSpent: storageMint.total_minted.toNumber()
+    };
+  }
+
+  async proposalStatistics () {
+    const proposalCounters = {
+      all: 0,
+      Active: 0,
+      Approved: 0,
+      Rejected: 0,
+      Expired: 0,
+      Slashed: 0,
+      Canceled: 0,
+      Vetoed: 0
+    };
+
+    const returnData = {
+      text: { ...proposalCounters },
+      spending: { ...proposalCounters },
+      workingGroups: { ...proposalCounters },
+      networkChanges: { ...proposalCounters },
+      all: { ...proposalCounters }
+    };
+
+    const proposals = await this.entriesByIds<ProposalId, ProposalDetails>(this.api.query.proposalsCodex.proposalDetailsByProposalId);
+    const proposalData = await this.api.query.proposalsEngine.proposals.multi<Proposal>(proposals.map(([id, _]) => id));
+
+    proposalData.map((proposal, index) => {
+      const proposalType = proposals[index][1].type;
+      let definedProposalType: 'text' | 'spending' | 'networkChanges' | 'workingGroups';
+      const proposalStatus = proposal.status.isOfType('Active') ? 'Active' as const : proposal.status.asType('Finalized').proposalStatus.type;
+
+      if (proposalType === 'Text') {
+        definedProposalType = 'text';
+      } else if (proposalType === 'Spending') {
+        definedProposalType = 'spending';
+      } else if (proposalType === 'RuntimeUpgrade' || proposalType === 'SetElectionParameters' || proposalType === 'SetValidatorCount') {
+        definedProposalType = 'networkChanges';
+      } else {
+        definedProposalType = 'workingGroups';
+      }
```

Comment:
I thought I mentioned this before (perhaps my comment didn't get submitted), but it would probably be better to not have a "fallback" like that. We most likely will be adding new proposal types in the future, so perhaps we can have an object like `{ [k in ProposalType]: GenereicProposalType }` to make sure all existing types are always mapped to a corresponding `'text' | 'spending' | 'networkChanges' | 'workingGroups'` (see ie.: `/joy-utils/src/consts/proposals.ts`)

Reply:
Wow. Such a nice solution!!

Should we make a separate one for `HISTORICAL_PROPOSALS` though or do we leave that as is?